teacher_class.rb

```diff
@@ -1 +1,8 @@
+class Teacher < Person
+  attr_reader :specialization
 
+  def initialize(id, age, specialization, name = 'Unknown', parent_permission: true)
+    super(id, age, name, parent_permission)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Since you are calling the `super` method here and the `parent_permission` method from the parent class is a **_keyword_** argument, you ought to pass it a value of the `parent_permission` parameter that is being created for the `Teacher` class. This will map the variables properly and the `super` method will use the conditions set in the `initialize` method of the `Person` class to be the same condition as those passed as parameters in the `super` method. Kindly implement this change so that you don't get these errors 👍 😎:

#### Hint;

```ruby
def initialize(id, age, specialization, name = 'Unknown', parent_permission: true)
  super(id, age, name, parent_permission: parent_permission)
  @specialization = specialization
end
```